File: android/app/src/main/java/com/br/app/deliverydelicatessen/model/Coupon.kt
package com.br.app.deliverydelicatessen.model

import java.util.ArrayList
import java.util.Date

data class Coupon(
    var code: String = "",
    var description: String = "",
    var value: Double = 0.00,
    var minimumValue: Double = 0.00,
    var establishmentId: Int = 0,
    var quantity: Int = 0,
    var id: Int = 0,
    var type: Boolean = false,
    var main: Boolean? = null,
    var active: Boolean = false,
    var general: Boolean = false,
    var initialDate: Date? = null,
    var finalDate: Date? = null,
    var clientId: String = ""
)

File: financial/Models/Coupon.cs

using System;
using System.ComponentModel.DataAnnotations.Schema;

namespace Models
{
    [Table("Coupon")]
    public class Coupon : BaseEntity
    {
        public string Code { get; set; }
        public string Description { get; set; }
        public bool Active { get; set; }
        public bool General { get; set; }
        public bool? Main { get; set; }
        public bool Type { get; set; }
        public int Quantity { get; set; }
        public decimal Value { get; set; }
        public decimal MinimumValue { get; set; }
        public DateTime InitialDate { get; set; }
        public DateTime FinalDate { get; set; }
        public int EstablishmentId { get; set; }

        public string UpdateApplicationUserId { get; set; }
        public DateTime CreateDate { get; set; }
        public DateTime? UpdateDate { get; set; }
        public string AspNetUsersId { get; set; }
        public string ClientId { get; set; }

    }
}
